File: Java/Implementations/Decorator/ColorDecorator.java
public class ColorDecorator extends DisplayDecorator {
	private Color color;

	public ColorDecorator(Color color, Display contained) {
		super(contained);
		this.color = color;
	}

	public String display() {
		return this.color.colorCode() + this.contained.display();
	}
}

File: Haskell/Implementations/Decorator/ColorDecorator.hs
module ColorDecorator
    ( ColorDecorator(..)
    ) where

import Color (Color(..))
import Display (Display(..))

data ColorDecorator c a = MkColorDecorator c a

instance (Display a, Color c) => Display (ColorDecorator c a) where
    display (MkColorDecorator color x) = colorCode color <> display x
